pkg/engine/engine.go

```diff
@@ -93,61 +95,62 @@ func (eng *Eng) RunTask(ctx context.Context, cfg *config.Task) error {
 
 // GetTask initializes and validates task
 func (eng *Eng) GetTask(cfg *config.Task) (Runnable, error) {
-	eng.mutex.Lock()
-	defer eng.mutex.Unlock()
+	//eng.mutex.Lock()
+	//defer eng.mutex.Unlock()
```

Comment:
Is this change intentional? Do we no longer required a mutex?

Reply:
This is temporary, I will revert it back.

Reply:
reverted.